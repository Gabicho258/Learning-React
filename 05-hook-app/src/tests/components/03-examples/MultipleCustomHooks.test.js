import React from "react";
import { shallow } from "enzyme";
import { MultipleCustomHooks } from "../../../components/03-examples/MultipleCustomHooks";
import { useFetch } from "../../../Hooks/useFetch";
import { useCounter } from "../../../Hooks/useCounter";

jest.mock("../../../Hooks/useFetch");
// jest.mock("../../../Hooks/useCounter");

describe("pruebas en MultipleCustomHooks", () => {
  //   useCounter.mockReturnValue({
  //     counter: 0,
  //     increment: () => {},
  //   });
  test("debe mostrarse correctamente", () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper).toMatchSnapshot();
  });
  test("debe mostrar la informacion", () => {
    useFetch.mockReturnValue({
      data: [{ author: "Gabo", quote: "ReactBest" }],
      loading: false,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper.find(".alert").exists()).toBe(false);
    expect(wrapper.find(".mb-0").text().trim()).toBe("ReactBest");
    expect(wrapper.find("footer").text().trim()).toBe("Gabo");
  });
});
