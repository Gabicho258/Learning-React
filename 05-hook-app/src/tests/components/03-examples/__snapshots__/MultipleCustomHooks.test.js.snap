// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en MultipleCustomHooks debe mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente Quote
  </button>
</Fragment>
`;
